frontend/src/features/leases/hooks/useLeaseDetail.ts

```diff
@@ -6,11 +6,8 @@ import { hideLoading, showLoading } from 'react-redux-loading-bar';
 import { toast } from 'react-toastify';
 import { logError } from 'store/slices/network/networkSlice';
 
-/**
- * hook that fetches the lease given the lease id.
- * @param leaseId
- */
-export const useLeaseDetail = (leaseId?: number) => {
+export function useLeaseDetail(leaseId: number) {
+  const [needsUpdate, setRefreshIndex] = useState<number>(0);
   const [lease, setLease] = useState<ILease>();
```

Comment:
if you were going to update the lease, you would return this setLease to the leasecontainer

Reply:
Not sure I follow. I tried this approach with the previous definition of useLeaseDetail but React complained about calling hooks in a callback.